File: codingame/community/theExperienceForCreatingPuzzles/theExperienceForCreatingPuzzles.m
#include <Foundation/Foundation.h>

int main(int argc, const char * argv[]) {
    int lvl;
    scanf("%d", &lvl);
    int needed;
    scanf("%d", &needed);
    int xp;
    scanf("%d", &xp);
    xp = xp * 300;
    
    while (xp >= needed) {
        lvl = lvl + 1;
        xp = xp - needed;
        needed = (lvl * sqrt(lvl) * 10);
        
    }
    
    printf("%d\n",lvl);
    printf("%d\n", needed - xp);
}

File: codingame/community/theExperienceForCreatingPuzzles/theExperienceForCreatingPuzzles.php
<?php
fscanf(STDIN, "%d",
    $lvl
);
fscanf(STDIN, "%d",
    $needed
);
fscanf(STDIN, "%d",
    $xp
);
$xp = $xp * 300;

while ($xp >= $needed) {
    $lvl = $lvl + 1;
    $xp = $xp - $needed;
    $needed = floor(($lvl * sqrt($lvl) * 10));
}
echo("$lvl\n");

print $needed - $xp;
?>